libsolidity/ast/Types.cpp

```diff
@@ -486,72 +699,110 @@ TypePointer IntegerConstantType::binaryOperatorResult(Token::Value _operator, Ty
 			break;
 		case Token::Mul:
 			value = m_value * other.m_value;
-			break;
+			break;			
 		case Token::Div:
 			if (other.m_value == 0)
 				return TypePointer();
-			value = m_value / other.m_value;
+			else
+				value = m_value / other.m_value;
 			break;
 		case Token::Mod:
 			if (other.m_value == 0)
 				return TypePointer();
-			value = m_value % other.m_value;
-			break;
+			else if (fixedPointType)
+			{
+				value = m_value;
+				if (value > other.m_value)
```

Comment:
`a % b = c <=> 0 <= c < |b| && \exist x \in \int: a = xb+c`
So what about something like `a - ((a/b).numerator() / (a/b).denominator()) * b`?
Concerning the sign of `a%b`, please follow the convention used by the EVM itself.